core/lib/multivm/src/versions/vm_refunds_enhancement/oracles/storage.rs

```diff
@@ -98,13 +107,17 @@ impl<S: WriteStorage, H: HistoryMode> StorageOracle<S, H> {
 
     pub fn read_value(&mut self, mut query: LogQuery) -> LogQuery {
         let key = triplet_to_storage_key(query.shard_id, query.address, query.key);
+
+        if !self.read_keys.inner().contains_key(&key) {
+            self.read_keys.insert(key, (), query.timestamp);
+        }
```

Comment:
i still think, that you can use just storage view.
And please remove it from all vms

Reply:
You mean removing the read_keys field and instead checking via the storage_view?
Like this?
`        if !self
            .storage
            .get_ptr()
            .borrow()
            .modified_storage_keys()
            .contains_key(&key)`

Reply:
Also do I removed it from other implementations apart from the ones in this PR? because its present in latest and 1_4_1 which are where I copied this from.

Reply:
<img width="901" alt="image" src="https://github.com/matter-labs/zksync-era/assets/8031374/8be3d4d4-b1ac-4e02-9d7d-bef829d3a308">
Storage view has `read_storage_keys` could we reuse it?